docs/rfcs/0002/readme.md

```diff
@@ -37,21 +34,98 @@ The proposal below might take a couple of read-throughs; I've also added a concr
+    3. the NNF storage requirements described above.
+3. The ADMINISTRATOR creates a corresponding _NNF Container Profile_ Kubernetes custom resource with the necessary NNF storage requirements and pod specification as described by the AUTHOR
+4. The USER who desires to use the application works with the AUTHOR and the related NNF Container Profile to understand the storage requirements
+5. The USER submits a WLM job with the #DW container directive variables populated
+6. WLM runs the workflow and drives it through the following stages...
+    1. `Proposal`: RABBIT validates the #DW container directive by comparing the supplied values to those listed in the NNF Container Profile. If the workflow fails to meet the requirements, the job fails
+    2. `PreRun`: RABBIT software:
+        1. creates a config map reflecting the storage requirements and any runtime parameters; this is provided to the container at the volume mount named `nnf-config`, if specified
+        2. duplicates the pod template specification from the Container Profile and patches the necessary Volumes and the config map. The spec is used as the basis for starting the necessary pods and containers
+    3. The containerized application(s) executes. The expected mounts are available per the requirements and celebration occurs. The pods continue to run until:
+       1. a pod completes successfully (any failed pods will be retried)
+       2. the max number of pod retries is hit (indicating failure on all retry attempts)
+          1. Note: retry limit is non-optional per Kubernetes configuration
+          2. If retries are not desired, this number could be set to 0 to disable any retry attempts
+    4. `PostRun`: RABBIT software:
+       1. marks the stage as `Ready` if the pods have all completed successfully. This includes a successful retry after preceding failures
+       2. starts a timer for any running pods. Once the timeout is hit, the pods will be killed and the workflow will indicate failure
+       3. leaves all pods around for log inspection
+
+### Container Assignment to Rabbit Nodes
+
+During `Proposal`, the USER must assign compute nodes for the container workflow. The assigned
+compute nodes determine which Rabbit nodes run the containers.
+
+### Communication Details
+
+The following subsections outline the proposed communication between the Rabbit nodes themselves and
+the Compute nodes.
+
+#### Rabbit-to-Rabbit Communication
+
+Each rabbit node can be reached via `<hostname>.<subdomain>` using DNS. The hostname is a
+combination of the workflow name and Rabbit node name. The workflow name is used for the subdomain.
+
+For example, a workflow name of `foo` that targets `rabbit-node2` would be `foo-rabbit-node2.foo`.
+
+Environment variables are provided to the container and ConfigMap for each rabbit that is targeted
+by the container workflow:
+
+```shell
+RABBIT_HOSTS=foo-rabbit-node2,foo-rabbit-node3
```

Comment:
We use NNF_ elsewhere, specifically NNF_NODE_NAME. Also wonder if we should include something that makes this obvious it's an NNF container variable. Maybe NNF_CONTAINER_(NODES | SUBDOMAIN | DOMAIN).

Reply:
Good call. Is it reasonable to still use rabbit in the node names (e.g. `foo-rabbit-node2`) though? We do use rabbit quite a bit in the document.

Reply:
Fine for now. Examples are much easier to change. We could stick with the `rabbit-node-X` / `compute-node-Y` pattern that is in a couple of entries in nnf-deploy's systems.yaml.

I'm terrible at naming things, so maybe check with others on their preferences